src/services/ILeaderboardManager.hh

```diff
@@ -26,8 +26,8 @@ public:
     virtual void AddLeaderboard(RA_Leaderboard&& lb) = 0;
     virtual size_t Count() const = 0;
     virtual const RA_Leaderboard& GetLB(size_t index) const = 0;
-    virtual RA_Leaderboard* FindLB(LeaderboardID nID) = 0;
-    const RA_Leaderboard* FindLB(LeaderboardID nID) const { return const_cast<ILeaderboardManager*>(this)->FindLB(nID); }
```

Comment:
Done. Implementation done in a static helper, and is called by both of them w/o `const_cast`.

Reply:
Your solution still requires overriding both virtual methods. I was proposing something like this:
```
public:
    RA_Leaderboard* FindLB(LeaderboardID nID) { return FindLBImpl(nID); }
    const RA_Leaderboard* FindLB(LeaderboardID nID) const { return FindLBImpl(nID); }
protected:
    virtual RA_Leaderboard* FindLBImpl(LeaderboardID nID) const;
```

Reply:
I can try to do that.

Reply:
I tried doing that but it doesn't work. The constness is fixed since it declared instead of picked/deduced. The helper has to be static because static functions can't be const qualified.